client/src/App.js

```diff
@@ -119,14 +120,48 @@ fetchMaterials = () => {
       });
 }
 
+fetchFacilityDetails = (facility_id) => {
+  //Fetches facilities that can service user selcted facility on this card
+  let url = `/api/facilities/earth911/facilities/${facility_id}`
+
+  let myPromise = fetch(url, {})
+                    .then((response) => {
+                      return response.json();
+                    })
+                    .then((facility_data) => {
+                      console.log(facility_data);
+                      this.setState({facilityDetails: facility_data.results.result[facility_id]});
+                    }).then((data) => {
+                      const myPromise = new Promise(function(resolve, reject) {
+                         resolve();
+                      });
+                      return myPromise
+                    });
+  return myPromise
```

Comment:
F.Y.I. 
I can see you wanted to do this function as a promise. You could have done that using `async/await`. A lot simpler than this block.